DBproject_part2 (2).py

```diff
@@ -0,0 +1,132 @@
+#My_project  imports -> function -> code
+def menu():
+    print("1. Save a new entry")
+    print("2. Search by ID")
+    print("3. Print ages average")
+    print("4. Print all names")
+    print("5. Print all IDs")
+    print("6. Print all entries")
+    print("7. Print entry by index")
+    print("8. Exit")
+
+def saveNewEntry(entries):
+    id_input = input("ID: ")
+    if not id_input.isdigit():
+        print("ERROR: Your ID can only contain numbers " + str(id_input) + " is not a number")
+        input("Press Enter to continue...")
+        return None
+    if id_input in entries:
+        print("ERROR: The ID " + str(id_input) + " already exists.")
+        input("Press Enter to continue...")            #creat function ToContinue 
+        return None
+    else: 
+        name_input = input("Name: ")
+    age_input = input("Age: ")
+    if not age_input.isdigit():
+        print(f"ERROR: Age must be a number. ['{age_input}'] is not valid.")
+        age_input = 0
+        input("Press Enter to continue...")
+        return None
+    else:
+        entries[id_input] = [name_input, age_input]
+        print("ID [" + str(id_input) + "] saved successfuly!")
+        input("Press Enter to continue...")
+        return int(age_input)
+
+def searchById(entries):
+    id_to_search = input("Enter the ID you looking for: ")
+    if not id_to_search.isdigit():
+        print(f"ERROR: ID must be a number. ['{id_to_search}'] is not valid.")
+    elif id_to_search not in entries:
+        print(f"ERROR: The ID ['{id_to_search}'] was not found.")
+    else:
+        return "ID found:" + entries[id_to_search]
+    
+#def calculatAgesAverage(entries,):
+
+def printAllNames(entries):
+    for name in entries.values():
+        name = entries[0]
+        if len(entries) == 0:
+            print("There's no entries yet, so no names to print.")
+            return
+        else:
+            print("All the names in the database: ")

```

Comment:
השורה הזו תודפס עבור כל איבר במילון. את בטוחה שזה מה שאת רוצה שיקרה?
All the names in the database:
תזכרי, זה בתוך לולאה - זה יקרה בכל איטרציה